### PR TITLE
✨ Improve unstructured serialisation

### PR DESCRIPTION
**Pull request by inteon**
Currently, we use a custom `NegotiatedSerializer` for unstructured serialisation. This PR proposes to use the client-go unstructured `NegotiatedSerializer` instead.
This serializer does try to deserialize all json blobs to unstructured structs, making it possible to use Watch together with this deserializer (see changes to `pkg/client/watch.go`).

### CODE REVIEW

File: pkg/client/apiutil/apimachinery.go [hunk 1]
```diff
@@ -153,19 +154,6 @@ func RESTClientForGVK(gvk schema.GroupVersionKind, isUnstructured bool, baseConf
 	return rest.RESTClientFor(createRestConfig(gvk, isUnstructured, baseConfig, codecs))
 }
 
-// serializerWithDecodedGVK is a CodecFactory that overrides the DecoderToVersion of a WithoutConversionCodecFactory
-// in order to avoid clearing the GVK from the decoded object.
-//
-// See https://github.com/kubernetes/kubernetes/issues/80609.
```
Comment (vincepri) [inline pkg/client/apiutil/apimachinery.go:159]: 🤔 Without this, is the GVK preserved? The issue linked here is still open.

Comment (inteon) [reply pkg/client/apiutil/apimachinery.go:159]: The watch does preserve the GVK value, I added a test to prove this https://github.com/kubernetes-sigs/controller-runtime/pull/2147/commits/fa6aa5463b0de0a82bd435f1664a95052e67599a.

**Review by vincepri — APPROVED** [review]
Thanks!

/approve
/lgtm

**Pull request merged**